### PR TITLE
Gui: functions to store/recall temporary working view moved from hardcoded to commands

### PR DESCRIPTION
**Pull request by 0penBrain**
This PR moves 2 functions used to store/recall a document-specific temporary view in 3D viewport : 
- from hardcoded in the navigation style base class
- to 2 normal commands that user can use as needed (especially customize shortcuts)
The hardcoded behavior was creating some very weird effect to user in specific conditions (see forum link)

The commands are added at the end of the `standard view` submenu in the `view` menu.
Default shortcuts are `Shift+End` and `End` respectively for store and recall.

image

https://forum.freecadweb.org/viewtopic.php?f=3&t=72082

### CODE REVIEW

**Comment by 0penBrain** [comment]
@chennes @luzpaz may you please check the English phrasing ?

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7525">pipeline status</a> for feature branch [PR_7525](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7525). Pipeline [650025803](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/650025803) was triggered at [96b3d98](https://github.com/FreeCAD/FreeCAD/pull/7525/commits/96b3d985f634007b44bc090651c34c5695845e76). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by wwmayer** [comment]
> may you please check the English phrasing ?

What about the terms "Save" and "Restore"?

File: src/Gui/CommandView.cpp [hunk 1]
```diff
@@ -3614,6 +3614,70 @@ Action * StdCmdSelBoundingBox::createAction()
...
+    sWhatsThis    = "Std_StoreWorkingView";
+    sAccel        = "Shift+End";
+    eType         = NoTransaction;
+}
+
+void StdStoreWorkingView::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    if (isActive()) {
+        auto view = static_cast<Gui::View3DInventor*>(Gui::getMainWindow()->activeWindow());
```
Comment (wwmayer) [inline src/Gui/CommandView.cpp:3638]: The static_cast relies on the correct implementation of isActive(). To be more robust you should use dynamic_cast and check the returned pointer.

```
void StdStoreWorkingView::activated(int iMsg)
{
    Q_UNUSED(iMsg);
    if (auto view = dynamic_cast<Gui::View3DInventor*>(Gui::getMainWindow()->activeWindow());) {
        view->getViewer()->saveHomePosition();
    }
}
```

**Comment by 0penBrain** [comment]
> What about the terms "Save" and "Restore"?

I'm not native speaker, but from my memories these are terms I often saw to express this kind of temporary+limited storage (like if you have some "slots" -- here only one --). "Save" and "restore" sounds to me more like external lasting storage.
But at the end really I don't care about wording. I'll put whatever someone may decide.

**Comment by berndhahnebach** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7525">pipeline status</a> for feature branch [PR_7525](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_7525). Pipeline [650031312](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/650031312) was triggered at [868ab42](https://github.com/FreeCAD/FreeCAD/pull/7525/commits/868ab42a87be34d998e68f905a49302e8cea3b96). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Comment by chennes** [comment]
There may be slight connotation differences between "Store"/"Recall" and "Save"/"Restore", but I think they are subtle enough to not worry about. Either phrasing would make sense to me as a user, I think. I'd probably leave the PR as you have it.

**Comment by luzpaz** [comment]
Agree with chennes's assessment

**Comment by adrianinsaval** [comment]
How is this different to the stuff in the "Freeze display" menu? (that one has a shit name though)

**Comment by adrianinsaval** [comment]
Freeze display is also temporary unless you manually save and then manually load in the next session. IMO there's no need to add a new couple of commands that do basically the same. In any case assign default shortcuts for the commands there and maybe change the wording.

The removal of the harcoded shortcuts should totally be merged though.

**Comment by 0penBrain** [comment]
@adrianinsaval The "freeze display" is really unpolished IMO (you can't change the shortcut, you can't name the view, ...).
My proposal would be to refactor the "freeze view" so frozen views are automatically saved to user config, and that you can also choose to export/import from/to a file.

**Comment by donovaly** [comment]
>> The commands are added at the end of the `standard view` submenu in the `view` menu. Default shortcuts are `Shift+End` and `End` respectively for store and recall.

Many thanks. Don't forget to add this new feature to the FreeCAD 1.0 wiki page.

**Comment by 0penBrain** [comment]
@donovaly Not a new feature actually. Just it was so hidden that probably was unknown. :)
Will put a word about that in the wiki. ;)

**Comment by donovaly** [comment]
> Just it was so hidden that probably was unknown. :) Will put a word about that in the wiki. ;)

Many thanks.
For an average user he sees now 2 more menu items.. Therefore the release notes should mention this.

Besides this, please also update the Wiki page describing this menu.

**Pull request merged**